include/mgp_mock.py

```diff
@@ -0,0 +1,1916 @@
+    def __init__(self, vertex):
+        if not isinstance(vertex, _mgp_mock.Vertex):
+            raise TypeError(f"Expected '_mgp_mock.Vertex', got '{type(vertex)}'")
+
+        self._vertex = vertex
+
+    def __deepcopy__(self, memo):
+        # In line with the Python API, this is the same as the shallow copy.
+        return Vertex(self._vertex)
+
+    def is_valid(self) -> bool:
+        """
+        Check if the vertex is in a valid context, i.e. if it may be used.
+
+        Returns:
+            A `bool` value that represents whether the vertex is in a valid context.
+
+        Examples:
+            ```vertex.is_valid()```
+        """
+        return self._vertex.is_valid()
+
+    def underlying_graph_is_mutable(self) -> bool:
+        """
+        Check if the underlying graph is mutable.
+
+        Returns:
+            A `bool` value that represents whether the graph is mutable.
+
+        Raises:
+            InvalidContextError: If the context is not valid.
+
+        Examples:
+            ```edge.underlying_graph_is_mutable()```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return self._vertex.underlying_graph_is_mutable()
+
+    @property
+    def id(self) -> VertexId:
+        """
+        Get the ID of the vertex.
+
+        Returns:
+            A `VertexId` representing the vertex’s ID.
+
+        Raises:
+            InvalidContextError: If vertex is out of context.
+
+        Examples:
+            ```vertex.id```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return self._vertex.id
+
+    @property
+    def labels(self) -> typing.Tuple[Label]:
+        """
+        Get the labels of the vertex.
+
+        Returns:
+            A tuple of `Label` instances representing individual labels.
+
+        Raises:
+            InvalidContextError: If vertex is out of context.
+            DeletedObjectError: If `Vertex` has been deleted.
+
+        Examples:
+            ```vertex.labels```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._vertex.is_deleted():
+            raise DeletedObjectError("Accessing deleted object.")
+
+        return tuple(Label(label) for label in self._vertex.labels)
+
+    def add_label(self, label: str) -> None:
+        """
+        Add the given label to the vertex.
+
+        Args:
+            label: The label (`str`) to be added.
+
+        Raises:
+            InvalidContextError: If `Vertex` is out of context.
+            ImmutableObjectError: If `Vertex` is immutable.
+            DeletedObjectError: If `Vertex` has been deleted.
+
+        Examples:
+            ```vertex.add_label(label)```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._vertex.is_deleted():
+            raise DeletedObjectError("Accessing deleted object.")
+
+        return self._vertex.add_label(label)
```

Comment:
Why return when underlying _mgp_mock doesn't return anything?

Reply:
The `mgp_mock.add_label` and `_mgp_mock.add_label` methods both return `None`, so there’s no harm in doing this.
In fact, the Python API does the same thing in `mgp.py`.